lib/extensions.js

```diff
@@ -80,7 +80,7 @@ module.exports = function(realmConstructor) {
             // result in sync rejecting the writes. `_waitForDownload` ensures that the session is kept
             // alive until our callback has returned, which prevents it from being torn down and recreated
             // when we close the schemaless Realm and open it with the correct schema.
-            if (!config.sync.fullSynchronization && config.schema === undefined) {
+            if (config.sync.fullSynchronized !== undefined && !config.sync.fullSynchronization && config.schema === undefined) {
```

Comment:
I think the check should be `config.sync.fullSynchronization === false` - otherwise this will get triggered even when full sync is `true`.